### PR TITLE
docs: more information on trie

### PR DESCRIPTION
**Pull request by Longarithm**
That should be a go-to page if you have any questions about storage.

There are indeed many pieces missing:
* how memtrie works
* how storage costs are structured
* how storage proof is generated

but it would take much more time to describe it all.

### CODE REVIEW

**Review by akhi3030 — APPROVED** [review]

File: docs/architecture/storage/flow.md [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
+are connected with black arrow.
+* Memtrie lookup needs current state of accounting cache to compute costs. When
+query completes, accounting cache is updated with memtrie nodes. So they are connected
+with bidirectional black arrow.
+
+<!-- https://docs.google.com/presentation/d/1_iU5GfznFDUMUNi_7szBRd5hDrjqBxr8ap7eTCK-lZA/edit#slide=id.p  -->
+![Diagram with read and write request flow](https://private-user-images.githubusercontent.com/8607261/382874468-c2a1a441-7eb4-46de-a727-473d60d2bb18.svg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA3NDQ3ODgsIm5iZiI6MTczMDc0NDQ4OCwicGF0aCI6Ii84NjA3MjYxLzM4Mjg3NDQ2OC1jMmExYTQ0MS03ZWI0LTQ2ZGUtYTcyNy00NzNkNjBkMmJiMTguc3ZnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMDRUMTgyMTI4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGY4ZmQ3MjA0YWI4ZWFmNGY5MTA5YzBlNTg0MzI4OTBjMDhkOGY0OTY1NDA5Zjk1OWRlODU4NDIxZGU5MDRmZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.mWX24TIYt8GRqHtg3zVA5XUhcYrBhWjmjO7vRD_D75w)
```
Comment (nagisa) [inline docs/architecture/storage/flow.md:19]: This link is a 404.

Comment (Longarithm) [reply docs/architecture/storage/flow.md]: What about now?

**Review by nagisa — APPROVED** [review]

File: docs/architecture/storage/primitives.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+* `ReceivedData`
+* `PostponedReceiptId`
+* `PendingDataCount`
+* `PostponedReceipt`
+* `DelayedReceiptIndices`
+* `DelayedReceipt`
+* `ContractData`
+* `YieldedPromiseQueueIndices`
+* `YieldedPromiseQueueEntries`
+* `PromiseYieldReceipt`
```
Comment (shreyan-gupta) [inline docs/architecture/storage/primitives.md:19]: Could you quickly add three more keys here to the list? `BufferedReceiptIndices`, `BufferedReceipt` and `BandwidthSchedulerState`

**Review by shreyan-gupta — APPROVED** [review]
Thanks!

**Pull request merged**